elements/lisk-p2p/test/unit/peer_pool.ts

```diff
@@ -73,6 +79,216 @@ describe('peerPool', () => {
 		});
 	});
 
+	describe('#filterPeersByCategory', () => {
+		const originalPeers = [...new Array(10).keys()].map(i => ({
+			netgroup: i,
+			latency: i,
+			responseRate: i % 2 ? 0 : 1,
+			connectTime: i,
+		}));
+
+		it('should protect peers with highest netgroup value when sorted by ascending', async () => {
+			const filteredPeers = filterPeersByCategory(originalPeers as any, {
+				category: PROTECTION_CATEGORY.NET_GROUP,
+				percentage: 0.2,
+				asc: true,
+			});
+
+			filteredPeers.forEach(peer => {
+				expect(peer.netgroup).to.be.greaterThan(1);
+			});
+		});
+
+		it('should protect peers with lowest latency value when sorted by descending', async () => {
+			const filteredPeers = filterPeersByCategory(originalPeers as any, {
+				category: PROTECTION_CATEGORY.LATENCY,
+				percentage: 0.2,
+				asc: false,
+			});
+
+			filteredPeers.forEach(peer => {
+				expect(peer.latency).to.be.lessThan(3);
+			});
+		});
+
+		it('should protect 2 peers with responseRate value of 1 when sorted by ascending', async () => {
+			const filteredPeers = filterPeersByCategory(originalPeers as any, {
+				category: PROTECTION_CATEGORY.RESPONSE_RATE,
+				percentage: 0.2,
+				asc: true,
+			});
+
+			expect(filteredPeers.filter(p => p.responseRate === 1).length).to.eql(2);
+		});
+
+		it('should protect peers with lowest connectTime value when sorted by descending', async () => {
```

Comment:
Highest connectTime is a more important to test case.

Reply:
`connectTime` is the exact time of connect, not `duration` of connection time.

Reply:
Ah nevermind then  ;p